src/Microsoft.ML.Data/Transforms/TypeConverting.cs

```diff
@@ -173,45 +173,47 @@ private static VersionInfo GetVersionInfo()
+            public readonly string Source;
+            public readonly string Name;
             public readonly DataKind OutputKind;
             public readonly KeyRange OutputKeyRange;
 
             /// <summary>
             /// Describes how the transformer handles one column pair.
             /// </summary>
-            /// <param name="input">Name of input column.</param>
-            /// <param name="output">Name of output column.</param>
+            /// <param name="source">Name of input column.</param>
+            /// <param name="name">Name of output column.</param>
             /// <param name="outputKind">The expected kind of the converted column.</param>
             /// <param name="outputKeyRange">New key range, if we work with key type.</param>
-            public ColumnInfo(string input, string output, DataKind outputKind, KeyRange outputKeyRange = null)
+            public ColumnInfo(string name, string source, DataKind outputKind, KeyRange outputKeyRange = null)
             {
-                Input = input;
-                Output = output;
+                Contracts.CheckNonWhiteSpace(name, nameof(name));
+
+                Source = source;
+                Name = name;
                 OutputKind = outputKind;
                 OutputKeyRange = outputKeyRange;
             }
         }
 
         private readonly ColumnInfo[] _columns;
 
-        private static (string input, string output)[] GetColumnPairs(ColumnInfo[] columns)
+        private static (string source, string name)[] GetColumnPairs(ColumnInfo[] columns)
         {
             Contracts.CheckNonEmpty(columns, nameof(columns));
-            return columns.Select(x => (x.Input, x.Output)).ToArray();
+            return columns.Select(x => (x.Source, x.Name)).ToArray();
         }
 
         /// <summary>
         /// Convinence constructor for simple one column case.
         /// </summary>
         /// <param name="env">Host Environment.</param>
-        /// <param name="inputColumn">Name of the output column.</param>
-        /// <param name="outputColumn">Name of the column to be transformed. If this is null '<paramref name="inputColumn"/>' will be used.</param>
+        /// <param name="source">Name of the column produced.</param>
+        /// <param name="name">Name of the column to be transformed. If this is null '<paramref name="source"/>' will be used.</param>
         /// <param name="outputKind">The expected type of the converted column.</param>
         /// <param name="outputKeyRange">New key range if we work with key type.</param>
-        public TypeConvertingTransformer(IHostEnvironment env, string inputColumn, string outputColumn, DataKind outputKind, KeyRange outputKeyRange = null)
-            : this(env, new ColumnInfo(inputColumn, outputColumn, outputKind, outputKeyRange))
+        public TypeConvertingTransformer(IHostEnvironment env, string source, string name, DataKind outputKind, KeyRange outputKeyRange = null)
```

Comment:
>string source, string name [](start = 63, length = 26)

This is not in proper order.

Reply:
I will tackle the transformers in the next pr, because i have to change all of them, they all derive from OneToOneTransformer. 

Agree, it is confusing because I did partial work, on renaming the args. 

---
In reply to: [248078355](https://github.com/dotnet/machinelearning/pull/2118#discussion_r248078355) [](ancestors = 248078355)